torch_geometric/nn/models/explainer.py

```diff
@@ -114,31 +114,268 @@ def forward(self, mask, *args):
+
+        if self.mask_type == 'node':
+            node_tensors = args[:self.num_node_types]
+            node_tensors = [mask.squeeze(0) for mask in node_tensors]
+            x_dict = dict(zip(self.node_types, node_tensors))
+            edge_index_dict = args[self.num_node_types]
+        elif self.mask_type == 'edge':
+            edge_mask_tensors = args[:self.num_edge_types]
+            x_dict = args[self.num_edge_types]
+            edge_index_dict = args[self.num_edge_types + 1]
+        else:
+            node_tensors = args[:self.num_node_types]
+            node_tensors = [mask.squeeze(0) for mask in node_tensors]
+            x_dict = dict(zip(self.node_types, node_tensors))
+            edge_mask_tensors = args[self.num_node_types:self.num_node_types +
+                                     self.num_edge_types]
+            edge_index_dict = args[self.num_node_types + self.num_edge_types]
+
+        if 'edge' in self.mask_type:
+            edge_mask_tensors = [mask.squeeze(0) for mask in edge_mask_tensors]
+            edge_mask_dict = dict(zip(self.edge_types, edge_mask_tensors))
+        else:
+            edge_mask_dict = None
+        return x_dict, edge_index_dict, edge_mask_dict
+
+    def forward(self, *args):
+        # Validate args:
+        if self.mask_type == "node":
+            assert len(args) >= self.num_node_types + 1
+            len_remaining_args = len(args) - (self.num_node_types + 1)
+        elif self.mask_type == "edge":
+            assert len(args) >= self.num_edge_types + 2
+            len_remaining_args = len(args) - (self.num_edge_types + 2)
+        else:
+            assert len(args) >= self.num_node_types + self.num_edge_types + 1
+            len_remaining_args = len(args) - (self.num_node_types +
+                                              self.num_edge_types + 1)
+
+        # Get main args:
+        (x_dict, edge_index_dict,
+         edge_mask_dict) = self._captum_data_to_hetero_data(*args)
+
+        if 'edge' in self.mask_type:
+            set_hetero_masks(self.model, edge_mask_dict, edge_index_dict)
+
+        if len_remaining_args > 0:
+            # If there are args other than `x_dict` and `edge_index_dict`
+            x = self.model(x_dict, edge_index_dict,
+                           *args[-len_remaining_args:])
+        else:
+            x = self.model(x_dict, edge_index_dict)
+
+        if 'edge' in self.mask_type:
+            clear_masks(self.model)
+
+        if self.output_idx is not None:
+            x = x[self.output_idx].unsqueeze(0)
+        return x
+
+
+def _to_edge_mask(edge_index: Tensor) -> Tensor:
+    num_edges = edge_index.shape[1]
+    return torch.ones(num_edges, requires_grad=True, device=edge_index.device)
+
+
+def _raise_on_invalid_mask_type(mask_type: str):
+    if mask_type not in ['node', 'edge', 'node_and_edge']:
+        raise ValueError(f"Invalid mask type (got {mask_type})")
+
+
+def to_captum_input(data: Union[HeteroData, Data], mask_type: str,
```

Comment:
I am not a big fan here of expecting `data` as input. Can we make this `x` and `edge_index` instead?

Reply:
Yeah i agree that makes more sense. The function signature will look like this
`to_captum_input(x: Union[Tensor, Dict[NodeType, Tensor]], edge_index: Union[Tensor, Dict[EdgeType, Tensor]], ...,)`
does it look ok?